sync/src/main/java/tech/pegasys/teku/sync/AttestationManager.java

```diff
@@ -40,42 +41,42 @@
 
   private final PendingPool<DelayableAttestation> pendingAttestations;
   private final FutureItems<DelayableAttestation> futureAttestations;
+  private final Consumer<Attestation> attestationPoolConsumer;
 
   AttestationManager(
       final EventBus eventBus,
       final ForkChoiceAttestationProcessor attestationProcessor,
       final PendingPool<DelayableAttestation> pendingAttestations,
-      final FutureItems<DelayableAttestation> futureAttestations) {
+      final FutureItems<DelayableAttestation> futureAttestations,
+      final Consumer<Attestation> attestationPoolConsumer) {
     this.eventBus = eventBus;
     this.attestationProcessor = attestationProcessor;
     this.pendingAttestations = pendingAttestations;
     this.futureAttestations = futureAttestations;
+    this.attestationPoolConsumer = attestationPoolConsumer;
   }
 
   public static AttestationManager create(
-      final EventBus eventBus,
-      final PendingPool<DelayableAttestation> pendingAttestations,
-      final FutureItems<DelayableAttestation> futureAttestations,
-      final ForkChoiceAttestationProcessor forkChoiceAttestationProcessor) {
+          final EventBus eventBus,
+          final PendingPool<DelayableAttestation> pendingAttestations,
+          final FutureItems<DelayableAttestation> futureAttestations,
+          final ForkChoiceAttestationProcessor forkChoiceAttestationProcessor,
+          final Consumer<Attestation> poolAttestationConsumer) {
     return new AttestationManager(
-        eventBus, forkChoiceAttestationProcessor, pendingAttestations, futureAttestations);
+        eventBus, forkChoiceAttestationProcessor, pendingAttestations, futureAttestations, poolAttestationConsumer);
   }
 
   @Subscribe
   @SuppressWarnings("unused")
   private void onGossipedAttestation(final Attestation attestation) {
-    processAttestation(
-        new DelayableAttestation(
-            attestation, () -> eventBus.post(new ProcessedAttestationEvent(attestation))));
+    processAttestation(new DelayableAttestation(attestation, attestationPoolConsumer));
   }
 
   @Subscribe
   @SuppressWarnings("unused")
   private void onAggregateAndProof(final SignedAggregateAndProof aggregateAndProof) {
     final Attestation aggregate = aggregateAndProof.getMessage().getAggregate();
-    processAttestation(
-        new DelayableAttestation(
-            aggregate, () -> eventBus.post(new ProcessedAggregateEvent(aggregate))));
+    processAttestation(new DelayableAttestation(aggregate, attestationPoolConsumer));
```

Comment:
I think we still need the distinction between posting a `ProcessedAttestationEvent` or a `ProcessedAggregateEvent`, just that in both cases we'd also add to the `AggregatingAttestationManager`. Or is nothing listening for those events?  If not, how do we wind up gossiping the attestation once it has been processed if it was initially delayed?

Reply:
Made a ticket for the attestations we should gossip after processing: https://pegasys1.atlassian.net/secure/RapidBoard.jspa?rapidView=67&modal=detail&selectedIssue=BC-450